tools/docker/frontend.py

```diff
@@ -1,18 +1,111 @@
 import argparse
-import subprocess
+import logging
 import os
+import re
+import subprocess
+import sys
+
+from six import iteritems
 
 here = os.path.abspath(os.path.dirname(__file__))
 wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
 
-def build(*args, **kwargs):
+logger = logging.getLogger()
+
+
+def build(tag="wpt:local", *args, **kwargs):
     subprocess.check_call(["docker",
                            "build",
                            "--pull",
-                           "--tag", "wpt:local",
+                           "--tag", tag,
                            here])
 
 
+def parser_push():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--tag", action="store",
+                        help="Tag to use (default is taken from .taskcluster.yml)")
+    parser.add_argument("--force", action="store_true",
+                        help="Ignore warnings and push anyway")
+    return parser
+
+
+def walk_yaml(root, target):
+    rv = []
+    if isinstance(root, list):
+        for value in root:
+            if isinstance(value, (dict, list)):
+                rv.extend(walk_yaml(value, target))
+    elif isinstance(root, dict):
+        for key, value in iteritems(root):
+            if isinstance(value, (dict, list)):
+                rv.extend(walk_yaml(value, target))
+            elif key == target:
+                rv.append(value)
+    return rv
+
+
+def read_image_name():
+    import yaml
+    with open(os.path.join(wpt_root, ".taskcluster.yml")) as f:
+        taskcluster_data = yaml.safe_load(f)
+    taskcluster_values = set(walk_yaml(taskcluster_data, "image"))
+    with open(os.path.join(wpt_root, "tools", "ci", "tc", "tasks", "test.yml")) as f:
+        test_data = yaml.safe_load(f)
+    tests_value = test_data["components"]["wpt-base"]["image"]
+    return taskcluster_values, tests_value
+
+
+def lookup_tag(tag):
+    import requests
+    org, repo_version = tag.split("/", 1)
+    repo, version = repo_version.rsplit(":", 1)
+    resp = requests.get("https://hub.docker.com/v2/repositories/webplatformtests/%s/tags/%s" %
+                        (repo, version))
```

Comment:
Should we use org instead of hard-coded "webplatformtests" here?
```suggestion
    resp = requests.get("https://hub.docker.com/v2/repositories/%s/%s/tags/%s" %
                        (org, repo, version))
```